### PR TITLE
Adding interface for the adagrad optimizer

### LINKED ISSUE #4916: Write Adagrad Op Python Wrapper

### PR DESCRIPTION
**Pull request by kexinzhao**
fixes #4916

### CODE REVIEW

File: python/paddle/v2/framework/optimizer.py [hunk 1]
```diff
@@ -272,3 +272,60 @@ def _append_optimize_op(self, block, param_and_grad):
...
 
         return momentum_op
+
+
+class AdagradOptimizer(Optimizer):
+    """Simple Adagrad optimizer with moment state
+    """
+    _moment_acc_str = "moment"
+
+    def __init__(self, learning_rate, epsilon):
```
Comment (abhinavarora) [inline python/paddle/v2/framework/optimizer.py]: You can give a default value to the epsilon.

Comment (abhinavarora) [reply python/paddle/v2/framework/optimizer.py]: Also if the original paper mentions a default learning rate value for the Adagrad optimizer, then you can add it too.

Comment (kexinzhao) [reply python/paddle/v2/framework/optimizer.py]: Done. The original paper does not specify a default learning rate. Only added default value to epsilon.
